django/core/signing.py

```diff
@@ -266,12 +279,13 @@ def unsign(self, value, max_age=None):
         """
         result = super().unsign(value)
         value, timestamp = result.rsplit(self.sep, 1)
-        timestamp = b62_decode(timestamp)
-        if max_age is not None:
-            if isinstance(max_age, datetime.timedelta):
-                max_age = max_age.total_seconds()
-            # Check timestamp is not older than max_age
-            age = time.time() - timestamp
-            if age > max_age:
-                raise SignatureExpired("Signature age %s > %s seconds" % (age, max_age))
+        self.payload_timestamp = b62_decode(timestamp)
+        check_signature_expiry(self.payload_timestamp, max_age=max_age)
+        return value
+
+    def unsign_object(self, signed_obj, max_age=None, return_timestamp=False, **kwargs):
+        # First verify the outer max_age boundary
+        value = super().unsign_object(signed_obj, max_age=max_age, **kwargs)
+        if return_timestamp:
```

Comment:
We do not do it currently but I think in the future Django will support type hints. Is it possible to annotate a function that changes the return values based on an input boolean? I think the answer is yes, but worth checking.

Reply:
Yes, in Python 3.10+ you can indicate multiple return types with the pipe operator:
```
def get_value(as_string: bool = False) -> int | str:
   result = 42
   return str(result) if as_string else result
```

For Django specifically, the Technical Board has stated [here](https://groups.google.com/g/django-developers/c/C_Phs05kL1Q/discussion):
> It is the view of the Django Technical Board that inline type annotations should not be added to Django at the current time.

Reply:
I think you can use [overload](https://docs.python.org/3/library/typing.html#overload) and literals to be more precise

Reply:
Yes, using `overload` sounds like a feasible approach.